### PR TITLE
Render Candidate specific redesigned Reasons for Rejection Component

### PR DESCRIPTION
**Pull request by cwrw**
## Context

The required design for the rejection reasons component for the provider/support interface is different than candidate's design.

## Changes proposed in this pull request

Add a candidate specific component which inherits the logic from the shared component however does not render the rejections in a table

## Guidance to review

Did I miss anything? 

## Link to Trello card

Before
Screenshot 2022-04-18 at 18 33 15

After
Screenshot 2022-04-18 at 18 32 28

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/components/candidate_interface/rejection_reasons/rejection_reasons_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            <% end %>
+          <% end %>
+        </ul>
+      <% else %>
+        <p><%= I18n.t("rejection_reasons.#{reason.id}.description") %></p>
+      <% end %>
+      <p><%= link_to_find_when_rejected_on_qualifications(application_choice) if render_link_to_find?(reason) %></p>
+    <% if editable? %>
+      <p class="app-rejection__actions">
+        <%= govuk_link_to 'Change', new_provider_interface_rejection_path(application_choice) %>
```
Comment (steventux) [inline app/components/candidate_interface/rejection_reasons/rejection_reasons_component.html.erb:29]: Is this link path correct ie. going from candidate to provider interface?

File: app/components/candidate_interface/rejection_reasons/rejection_reasons_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<% reasons.selected_reasons.each do |reason| %>
+  <div class="app-rejection">
```
Comment (steventux) [inline app/components/candidate_interface/rejection_reasons/rejection_reasons_component.html.erb:2]: It's a shame that we can't reuse https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/components/shared/rejection_reasons/rejection_reasons_component.html.erb as there are only minor design differences, not for this PR but perhaps we should ask design to consolidate the two sets of styles as we're doing a lot of conditional work to deliver two tantalizingly similar designs.

**Review by steventux — COMMENTED** [review]
Looks good, only one question about the change link in the candidate interface component.

File: app/components/candidate_interface/rejection_reasons/rejection_reasons_component.html.erb (hunk 1 above)
Comment (cwrw) [reply app/components/candidate_interface/rejection_reasons/rejection_reasons_component.html.erb]: addressed in https://github.com/DFE-Digital/apply-for-teacher-training/pull/6811/commits/74b6a97577c16b944fdd231265691443a23bdb6e

**Review by steventux — APPROVED** [review]
:+1:

**Pull request merged**